edge-modules/ModuleRestarter/src/Program.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace ModuleRestarter
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    class Program
+    {
+        static readonly ILogger Log = Logger.Factory.CreateLogger<Program>();
+        static readonly string ServiceClientConnectionString = Settings.Current.ServiceClientConnectionString;
+        static readonly string DeviceId = Settings.Current.DeviceId;
+        static readonly string DesiredModulesToRestartCSV = Settings.Current.DesiredModulesToRestartCSV;
+        static readonly int RandomRestartIntervalInMins = Settings.Current.RandomRestartIntervalInMins;
+
+        public static int Main() => MainAsync().Result;
+
+        static async Task<int> MainAsync()
+        {
+            Log.LogInformation($"Starting module restarter with the following settings:\r\n{Settings.Current}");
+
+            if (string.IsNullOrEmpty(DesiredModulesToRestartCSV))

```

Comment:
```suggestion
            if (string.IsNullOrWhiteSpace(DesiredModulesToRestartCSV))
```

Reply:
Even safer, you should parse here to ensure that you got non-empty list of module names (what happen if ",,," is given).

Reply:
is it ok to restart edgeAgent?

Reply:
I think we should not block this. If the user wants to restart edgeAgent, they should be able to.

Reply:
No exception gets thrown for repeated commas, so I will do the split() first then remove all empty strings. This edge case would happen by accident, so it's better not to halt the module restarting.